src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/InstanceNameTest/InstanceNameTest.cs

```diff
@@ -231,6 +230,42 @@ private static bool IsValidInstance(string browserHostName, string instanceName)
             return response != null && response.Length > 0;
         }
 
+        private static int GetNamedInstancePortNumberFromSqlBrowser(string connectionString)
+        {
+            SqlConnectionStringBuilder builder = new(connectionString);
+
+            string hostname = "";
+            string instanceName = "";
+            int port = 0;
+
+            DataTestUtility.ParseDataSource(builder.DataSource, out hostname, out _, out instanceName);
```

Comment:
I didn't notice this here either. You should Assert.Fail here if the datasource is not valid.
bool isDataSourceValid = blah
Assert.True(isDataSourceValid, "DataSource is invalid")

Reply:
The suggestion was applied.